### PR TITLE
fix creating list with JIT disabled

### PR DESCRIPTION
**Pull request by esc**
When using NUMBA_DISABLE_JIT=1 we still want `listobject.new_list` to
work, for example to measure coverage and such things.

Fixes https://github.com/numba/numba/issues/4905

### CODE REVIEW

File: numba/tests/test_listobject.py [hunk 1]
```diff
@@ -36,6 +36,10 @@ def foo(n):
         for i in (0, 1, 2, 100):
             self.assertEqual(foo(i), i)
 
+    def test_list_create_no_jit(self):
```
Comment (stuartarchibald) [inline numba/tests/test_listobject.py]: Whilst this checks the code path its not actually demonstrating the use of `NUMBA_DISABLE_JIT=1` ?

**Comment by seibert** [comment]
Following on Stuart's note, it looks like we have a test pattern that will let you verify with DISABLE_JIT enabled:

https://github.com/numba/numba/blob/92dbdb90e34402f99d158c035f5a1806378161d3/numba/tests/test_debug.py#L184-L188

**Comment by esc** [comment]
@seibert thanks for the feedback, I have implemented the tests accordingly and made them pass too.

**Comment by seibert** [comment]
Looks good.  Just waiting for CI to finish.

**Comment by esc** [comment]
No, it is still missing the test for just `List()`.

**Comment by esc** [comment]
hmmm, I think making `List()` return a `list` may require a metaclass

**Comment by seibert** [comment]
I think you can do it just by implementing `__new__()`.  Something like this might work:
```
     def __new__(cls, *args):
        if DISABLE_JIT:
            return object.__new__(list)
        else:       
             return object.__new__(cls)
```

**Comment by esc** [comment]
@seibert let me try that..

**Comment by esc** [comment]
The solution that passes all the tests is:

```
    def __new__(cls, lsttype=None, meminfo=None, allocated=None):
        if config.DISABLE_JIT:
            return list.__new__(list)
        else:
            return object.__new__(cls)
```

**Comment by esc** [comment]
But I am not sure a) what happens to the args and b) if this will fly on Py2

**Comment by esc** [comment]
pushed and waiting for CI

**Comment by seibert** [comment]
Looks good!  ready to merge

**Pull request merged**